lib/register.coffee

```diff
@@ -15,37 +15,38 @@ limitations under the License.
 ###
 
 Promise = require('bluebird')
+postAsync = Promise.promisify(require('request').post, multiArgs: true)
 _ = require('lodash')
 
+randomHexChar = ->
+	return _.random(15).toString(16)
+
 ###*
-# @summary Generate a device UUID
+# @summary Generate a random key, useful for both uuid and api key.
 # @function
 # @public
 #
 # @description
 # This function allows promise style if the callback is omitted.
 #
-# @param {Function} callback - callback (error, uuid)
+# @param {Function} callback - callback (error, randomKey)
 #
 # @example
-# deviceRegister.generateUUID (error, uuid) ->
-# 	throw error if error?
-#	# uuid is a generated UUID that can be used for registering
-# 	console.log(uuid)
+# deviceRegister.generateUniqueKey (error, randomKey) ->
+#  	throw error if error?
+# 	# randomKey is a randomly generated key that can be used as either a uuid or an api key
+# 	console.log(randomKey)
 ###
-exports.generateUUID = (callback) ->
-
-	# I'd be nice if the UUID matched the output of a SHA-256 function,
-	# but although the length limit of the CN attribute in a X.509
+exports.generateUniqueKey = (callback) ->
```

Comment:
Can we use node-randomstring? it does the right things. https://github.com/klughammer/node-randomstring